drivers/npmx/npmx_shell.c

```diff
@@ -416,42 +416,90 @@ static bool check_pin_configuration_correctness(const struct shell *shell, int8_
 	return true;
 }
 
-static int cmd_adc_meas_vbat_get(const struct shell *shell, size_t argc, char **argv)
+static int adc_meas_get(const struct shell *shell, npmx_adc_meas_t adc_meas,
+			npmx_adc_task_t adc_task)
 {
-	ARG_UNUSED(argc);
-	ARG_UNUSED(argv);
-
 	npmx_adc_t *adc_instance = adc_instance_get(shell);
 	if (adc_instance == NULL) {
 		return 0;
 	}
 
-	npmx_error_t err_code = npmx_adc_task_trigger(adc_instance, NPMX_ADC_TASK_SINGLE_SHOT_VBAT);
+	npmx_error_t err_code = npmx_adc_task_trigger(adc_instance, adc_task);
 	if (!check_error_code(shell, err_code)) {
 		shell_error(shell, "Error: unable to trigger the measurement.");
 		return 0;
 	}
 
-	int32_t voltage_mv;
+	int32_t value;
 	bool meas_get = false;
 	while (!meas_get) {
-		err_code = npmx_adc_meas_check(adc_instance, NPMX_ADC_MEAS_VBAT, &meas_get);
+		err_code = npmx_adc_meas_check(adc_instance, adc_meas, &meas_get);
 		if (!check_error_code(shell, err_code)) {
 			print_get_error(shell, "measurement status");
 			return 0;
 		}
 	}
 
-	err_code = npmx_adc_meas_get(adc_instance, NPMX_ADC_MEAS_VBAT, &voltage_mv);
+	err_code = npmx_adc_meas_get(adc_instance, adc_meas, &value);
 	if (!check_error_code(shell, err_code)) {
 		print_get_error(shell, "measurement");
 		return 0;
 	}
 
-	print_value(shell, voltage_mv, UNIT_TYPE_MILLIVOLT);
+	unit_type_t unit_type;
+	switch (adc_meas) {
+	case NPMX_ADC_MEAS_VBAT:
+		unit_type = UNIT_TYPE_MILLIVOLT;
+		break;
+	case NPMX_ADC_MEAS_VBAT2_IBAT:
+		unit_type = UNIT_TYPE_MILLIAMPERE;
+		break;
+	default:
+		return 0;
+	}
+
+	print_value(shell, value, unit_type);
 	return 0;
 }
 
+static int cmd_adc_meas_ibat_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	npmx_adc_t *adc_instance = adc_instance_get(shell);
+	if (adc_instance == NULL) {
+		return 0;
+	}
+
+	bool enable;
+	npmx_error_t err_code = npmx_adc_ibat_meas_enable_check(adc_instance, &enable);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "charging current enable");
+		return 0;
+	}
+
+	if (!enable) {
+		/* Enable auto measurement of the battery current after
+		 * the battery voltage measurement. */
+		err_code = npmx_adc_ibat_meas_enable_set(adc_instance, true);
+		if (!check_error_code(shell, err_code)) {
+			print_set_error(shell, "charging current enable");
+			return 0;
+		}
```

Comment:
Should it be disabled after getting the measurement?

Reply:
It is not required by an IPS.